docs/sdks/uid2-sdk-ref-python.md

```diff
@@ -90,62 +97,101 @@ Available information returned through the SDK is outlined in the following tabl
         print(err)
-      ``` 
-<!-- Alternative to the above for EUID:
-      from uid2_client import encrypt
-      from uid2_client.identity_scope import IdentityScope
-      
-        try:
-         identity_scope = IdentityScope.UID2  # or IdentityScope.EUID
-         encrypted_data = encrypt(raw_uid, identity_scope, keys)
-         #send encrypted_data to receiver
-      except EncryptionError as err:
-        #check for failure reason
-        print(err) -->
+      ```
 
 4. Receivers:
    1. Call the `decrypt` function. Then, if decryption succeeded, use the raw UID2:
     
       ```python
-      from uid2_client import decrypt
       try:
-        result = decrypt(ad_token, keys)
-        #use successfully decrypted result.uid2
+        result = client.decrypt(ad_token)
+        # use successfully decrypted result.uid2
       except EncryptionError as err:
-        #check for failure reason
+        # check for failure reason
         print(err)
       ```
 
+## Usage for Publishers
+
+1. Create an instance of Uid2PublisherClient
+
+   `client = Uid2PublisherClient(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY)`
+
+2. Call a function that takes the user's email address or phone number as input and generates a `TokenGenerateResponse` object. The following example uses an email address:
+
+   `token_generate_response = client.generate_token(TokenGenerateInput.from_email(emailAddress).do_not_generate_tokens_for_opted_out())`
+
+   >IMPORTANT: Be sure to call this function only when you have obtained legal basis to convert the user’s [directly identifying information (DII)](https://unifiedid.com/docs/ref-info/glossary-uid#gl-dii) to UID2 tokens for targeted advertising.
+
+   >`do_not_generate_tokens_for_opted_out()` applies `policy=1` in the [/token/generate](https://unifiedid.com/docs/endpoints/post-token-generate#token-generation-policy) call. Without this, `policy` is omitted to maintain backwards compatibility.
+
+### Standard Integration
+
+If you're using standard integration (client and server) (see [UID2 SDK for JavaScript Integration Guide](https://unifiedid.com/docs/guides/publisher-client-side)), follow this step:
+
+* Send this identity as a JSON string back to the client (to use in the [identity field](https://unifiedid.com/docs/sdks/client-side-identity#initopts-object-void)) using the following:
+
+  `token_generate_response.get_identity_json_string()` //Note: this method returns `None` if the user has opted out, so be sure to handle that case.
+
+### Server-Only Integration
+
+If you're using server-only integration (see [Publisher Integration Guide, Server-Only](https://unifiedid.com/docs/guides/custom-publisher-integration)):
+
+1. Store this identity as a JSON string in the user's session, using the `token_generate_response.get_identity_json_string()` function. This method returns `None` if the user has opted out, so be sure to handle that case.
+2. To retrieve the user's UID2 token, use:
```

Comment:
2. To retrieve the user's UID2 token, use the following:
(just added a couple of words -- doc style)

Reply:
fixed this